tutorials/cloud_access/euclid-cloud-access.md

```diff
@@ -0,0 +1,317 @@
+
++++
+
+## 2. Imports
+- `s3fs` for browsing S3 buckets
+- `astropy` for handling coordinates, units, FITS I/O, tables, images, etc.
+- `astroquery` for querying Euclid data products from IRSA 
+- `matplotlib` for visualization
+- `json` for decoding JSON strings
+
+```{code-cell} ipython3
+# Uncomment the next line to install dependencies if needed.
+# !pip install s3fs astropy astroquery matploltib
+```
+
+```{code-cell} ipython3
+import s3fs
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+from astropy.visualization import ImageNormalize, PercentileInterval, AsinhStretch
+from astropy.io import fits
+from astropy.nddata import Cutout2D
+from astropy.wcs import WCS
+from astropy.table import Table
+from astroquery.ipac.irsa import Irsa
+from matplotlib import pyplot as plt
+import json
+```
+
+## 3. Browse Euclid Q1 cloud-hosted data
+
+```{code-cell} ipython3
+BUCKET_NAME = 'nasa-irsa-euclid-q1'
+```
+
+[s3fs](https://s3fs.readthedocs.io/en/latest/) provides a filesystem-like python interface for AWS S3 buckets. First we create a s3 client:
+
+```{code-cell} ipython3
+s3 = s3fs.S3FileSystem(anon=True)
+```
+
+Then we list the `q1` directory that contains Euclid Q1 data products:
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1')
+```
+
+Let's navigate to MER images (available as FITS files):
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1/MER')[:10] # ls only top 10 to limit the long output
+```
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1/MER/102018211') # pick any tile ID from above
+```
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1/MER/102018211/VIS') # pick any instrument from above
+```
+
+As per "Browsable Directories" section in [user guide](https://irsa.ipac.caltech.edu/data/Euclid/docs/euclid_archive_at_irsa_user_guide.pdf), we need `MER/{tile_id}/{instrument}/EUC_MER_BGSUB-MOSAIC*.fits` for displaying background-subtracted mosiac images. But these images are stored under TILE IDs so first we need to find TILE ID for a coordinate search we are interested in. We will use astroquery (in next section) to retrieve FITS file paths for our coordinates by doing spatial search.
+
++++
+
+## 4. Do a spatial search for MER mosaics
+
+Pick a target and search radius:
+
+```{code-cell} ipython3
+target_name = 'TYC 4429-1677-1'
+coord = SkyCoord.from_name(target_name)
+search_radius = 10 * u.arcsec
+```
+
+List all Simple Image Access (SIA) collections for IRSA:
+
+```{code-cell} ipython3
+collections = Irsa.list_collections(servicetype='SIA')
+len(collections)
+```
+
+Filter to only those containing "euclid":
+
+```{code-cell} ipython3
+collections[['euclid' in v for v in collections['collection']]]
```

Comment:
fyi, this and similars in the other notebooks already triggered this feature request :)

https://github.com/astropy/astroquery/issues/3254